doc/coq.rst

```diff
@@ -0,0 +1,392 @@
+  is supported, but in this case the ``Coq`` prefix has been made available in a
+  qualified way, since :ref:`Coq lang 0.2<coq-lang>`.
+
+- You can enable the production of Coq's native compiler object files by setting
+  ``<coq_native_mode>`` to ``native``. This passes ``-native-compiler on`` to
+  Coq and install the corresponding object files under ``.coq-native``, when in
+  the ``release`` profile. The regular ``dev`` profile skips native compilation
+  to make the build faster. This has been available since :ref:`Coq lang
+  0.3<coq-lang>`.
+
+  Please note: support for ``native_compute`` is **experimental** and requires a
+  version of Coq later than 8.12.1. Furthermore, dependent libraries *must* be
+  built with the ``(mode native)`` enabled. In Addition to that, Coq must be
+  configured to support native compilation. Dune explicitly disables the
+  generation of native compilation objects when ``(mode vo)`` is enabled,
+  irrespective of the configuration of Coq. This will be improved in the future.
+
+.. _include-subdirs-coq:
+
+Recursive Qualification of Modules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you add:
+
+.. code:: scheme
+
+    (include_subdirs qualified)
+
+to a :ref:`dune file<dune-files>`, Dune considers all the modules in the
+directory and its subdirectories, adding a prefix to the module name in the
+usual Coq style for subdirectories. For example, file ``A/b/C.v`` becomes the
+module ``A.b.C``.
+
+Limitations
+~~~~~~~~~~~
+
+- ``.v`` files always depend on the native OCaml version of the Coq binary and
+  its plugins, unless the natively compiled versions are missing.
+
+.. _limitation-mlpack:
+
+- A ``foo.mlpack`` file must the present in directories of locally defined
+  plugins for things to work. This is a limitation of ``coqdep``.
+
+- Building a theory and a plugin in the same directory can lead to issues with
+  the presence of the META file. We recommend the following:
+  - A separate directory for the files of each :ref:`coq-theory` stanza defined.
+  - A separate directory for source files of a plugin.
+
+.. _coq-lang:
+
+Coq Language Version
+~~~~~~~~~~~~~~~~~~~~
+
+The Coq lang can be modified by adding the following to a :ref:`dune-project`
+file:
+
+.. code:: scheme
+
+    (using coq 0.3)
+
+The supported Coq language versions (not the version of Coq) are:
+
+- ``0.1``: Basic Coq theory support.
+- ``0.2``: Support for the ``theories`` field and composition of theories in the
+  same scope,
+- ``0.3``: Support for ``(mode native)`` requires Coq >= 8.10 (and Dune >= 2.9
+  for Coq >= 8.14).
+
+.. _coq-lang-1.0:
+
+Coq Language Version 1.0
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Guarantees with respect to stability are not yet provided. However, as the
+development of features progresses, we hope to reach ``1.0`` soon. The ``1.0``
+version of Coq lang will commit to a stable set of functionality. All the
+features below are expected to reach ``1.0`` unchanged or minimally modified.
```

Comment:
```suggestion
``1.0`` is still under development, so you may encounter issues using it.
There may be further changes to the features described below.
```

Think that's all the user needs to know (if I've stated the facts correctly).

Reply:
I think it may make sense to state that 1.0 will provide strong compat guarantees, so the user knows what to expect if they follow along. Also, I think it is very important to note that 0.3 etc... don't provide a lot of guaranttes. Not sure the reduced version conveys these points in full.

Reply:
Up to you guys, but I think compatibility is the default for Coq development if we don't state otherwise, plus we give deprecation warnings & try to make it easy to migrate to newer versions.  I'd hope that applies to Coq related parts of dinner, too.